File: Resueltos_por_competidor/Juan_Martinez/Dragons.cpp
/*
 * Autor: Juan Martinez
 * Problema: Dragons (230A)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/230/A
 * Difficulty: 1000
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	int s,n, t1, t2;
    cin>>s>>n;
    vector<pair<int,int>> drag(n);
    for(int i = 0; i < n; i++) {
        cin>>t1>>t2;
        drag[i] = {t1, t2};
    }
    sort(drag.begin(), drag.end());
    bool c = true;
    for(auto [f, b] : drag){
        if(s <= f) {
            c = false;
            break;
        } else {
            s += b;
        }
    }
    if(c) cout<<"YES"<<endl;
    else cout<<"NO"<<endl;
}

File: Resueltos_por_competidor/Martin_Peña/Dragons.java
/*
* Autor: pgms1103-afk
* Problema: A. Dragons
* Juez Online: Codeforces
* Veredicto: Accepted
* URL: https://codeforces.com/problemset/problem/230/A
*/
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;

public class Dragons {

	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

	public static void main(String[] args) throws NumberFormatException, IOException {

		String[] datos = br.readLine().split(" ");
		int s = Integer.parseInt(datos[0]);
		int n = Integer.parseInt(datos[1]);

		ArrayList<Integer> fuerzaDragon = new ArrayList<>();
		ArrayList<Integer> bonus = new ArrayList<>();
		for (int i = 0; i < n; i++) {
			String[] datos2 = br.readLine().split(" ");
			fuerzaDragon.add(Integer.parseInt(datos2[0]));
			bonus.add(Integer.parseInt(datos2[1]));
		}
		
		boolean bandera = true;
		while(bandera) {
			for (int i = 0; i < fuerzaDragon.size(); i++) {
				if(s > fuerzaDragon.get(i)) {
					s = s + bonus.get(i);
					fuerzaDragon.remove(i);
					bonus.remove(i);
					i--;
				}
			}
			if (fuerzaDragon.size() == 0) {
				break;
			}else {
				for (int j = 0; j < fuerzaDragon.size(); j++) {
					if(s > fuerzaDragon.get(j) && fuerzaDragon.size() != 0) {
						bandera = true;
						break;
					}else {
						bandera = false;
					}
				}
			}
			
		}
		if(fuerzaDragon.size() == 0) {
			System.out.println("YES");
		}else {
			System.out.println("NO");
		}

	}
}
